src/io/flutter/run/LaunchState.java

```diff
@@ -120,9 +121,21 @@ protected RunContentDescriptor launch(@NotNull ExecutionEnvironment env) throws
     final Project project = getEnvironment().getProject();
     final FlutterDevice device = DeviceService.getInstance(project).getSelectedDevice();
 
+    // If the device is null and the project is not a flutter web project, show a message that a device is required and return null.
```

Comment:
Nit: make the comment explain that flutter web doesn't yet support devices so this tweak is needed. As is the comment explains what happens which is visible from the code but not why that is a good thing.

Reply:
done.